libraries/freertos_plus/standard/freertos_plus_tcp/source/FreeRTOS_DNS.c

```diff
@@ -659,25 +746,27 @@ static const DNSMessage_t xDefaultPartDNSHeader =
 		( *pucStart )--;
 
 		pucStart = pucByte;
-
 	} while( *pucByte != 0x00 );
 
 	/* Finish off the record. */
 
-	pxTail = (DNSTail_t *)( pucByte + 1 );
+	pxTail = ( DNSTail_t * ) ( pucByte + 1 );
 
-	vSetField16( pxTail, DNSTail_t, usType, dnsTYPE_A_HOST );	/* Type A: host */
-	vSetField16( pxTail, DNSTail_t, usClass, dnsCLASS_IN );	/* 1: Class IN */
+	vSetField16( pxTail, DNSTail_t, usType, dnsTYPE_A_HOST ); /* Type A: host */
+	vSetField16( pxTail, DNSTail_t, usClass, dnsCLASS_IN );   /* 1: Class IN */
 
 	/* Return the total size of the generated message, which is the space from
 	the last written byte to the beginning of the buffer. */
 	return ( ( uint32_t ) pucByte - ( uint32_t ) pucUDPPayloadBuffer + 1 ) + sizeof( *pxTail );
 }
 /*-----------------------------------------------------------*/
 
-#if( ipconfigUSE_DNS_CACHE == 1 )
+#if( ipconfigUSE_DNS_CACHE == 1 ) || ( ipconfigDNS_USE_CALLBACKS != 0 )
```

Comment:
You're right, it is odd. I will change it to:

    #if( ipconfigUSE_DNS_CACHE == 1 ) || ( ipconfigDNS_USE_CALLBACKS == 1 )

Background: config macro's within FreeRTOS sources are mostly defined as 0 or 1. You will rarely see `#ifdef`'s. When a macro is not defined, it will always get a proper default value.

In logical expressions, there is a tendency to check if values are zero versus non-zero:

	#if( ipconfigDNS_USE_CALLBACKS != 0 )

	if( xFoo != pdFALSE )

But as `ipconfigDNS_USE_CALLBACKS` will either be 0 or 1, I will change the tests to `ipconfigDNS_USE_CALLBACKS == 1`.